### DISCUSSION TITLE
Testing pinia store actions with async/await calls

### DISCUSSION THREAD

**Discussion by Nallamachu** [vuejs/pinia#2240] [Help and Questions]
### What problem is this solving

In my project we are using PINIA for state and store management with the other technologies like Vue3+Vuetify3+Vite technologies.

We have done the API integration through PINIA actions with async/await calls. We are struggling to write test cases for the same PINIA action functions. Jest 29 is the testing library which we have included in our project.

I was also gone through the [PINIA testing documentation] (https://pinia.vuejs.org/cookbook/testing.html#testing-stores) but could not find the exact use case of what we need. So, I would like to request to include the possibility of writing test cases for PINIA store actions with async/await support.

### Proposed solution

Third party API testing through PINIA store actions. As of now, I don't have any proposed solutions for the same. I am seeking for help.

**Comment by posva** [comment]
Moving to a question as it’s more appropriate 🙂
